tests/app/main/test_frameworks.py

```diff
@@ -2805,14 +2810,188 @@ def test_get_with_with_previous_answers(self, data_api_client):
             doc = html.fromstring(res.get_data(as_text=True))
             assert len(doc.xpath('//input[@id="input-PR1-no"]/@checked')) == 1
 
-    def test_post_valid_data(self, data_api_client):
+    def test_get_with_with_prefilled_answers(self, data_api_client):
         with self.app.test_client():
             self.login()
+            # First call is for the current framework; second call for the framework to pre-fill from
+            data_api_client.get_framework.side_effect = [
+                self.framework(slug='g-cloud-9', name='G-Cloud 9', status='open'),
+                self.framework(slug='digital-outcomes-and-specialists-2', name='Digital Stuff 2', status='live'),
+            ]
+            # Current framework application information
+            data_api_client.get_supplier_framework_info.return_value = self.supplier_framework(
+                framework_slug="g-cloud-9",
+                declaration={"status": "started"},
+                prefill_declaration_from_framework_slug="digital-outcomes-and-specialists-2"
+            )
+            # The previous declaration to prefill from
+            data_api_client.get_supplier_declaration.return_value = {
```

Comment:
It feels to me like testing _all_ the API calls made by a view could fall into the category of "I know too much about the inside of the function under test" kind of testing.